api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -1036,6 +1036,102 @@ func TestAzureManagedControlPlane_ValidateUpdate(t *testing.T) {
 			amcp:    createAzureManagedControlPlane("192.168.0.10", "1.999.9", generateSSHPublicKey(true)),
 			wantErr: true,
 		},
+		{
+			name: "AzureManagedControlPlane AddonProfiles is mutable",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					Version: "v1.18.0",
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					Version: "v1.18.0",
+					AddonProfiles: []AddonProfile{
+						{
+							Name:    "first-addon-profile",
+							Enabled: true,
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "AzureManagedControlPlane AddonProfiles can be disabled",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AddonProfiles: []AddonProfile{
+						{
+							Name:    "first-addon-profile",
+							Enabled: true,
+						},
+					},
+					Version: "v1.18.0",
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					Version: "v1.18.0",
+					AddonProfiles: []AddonProfile{
+						{
+							Name:    "first-addon-profile",
+							Enabled: false,
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "AzureManagedControlPlane AddonProfiles cannot update to empty array",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AddonProfiles: []AddonProfile{
+						{
+							Name:    "first-addon-profile",
+							Enabled: true,
+						},
+					},
+					Version: "v1.18.0",
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					Version: "v1.18.0",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "AzureManagedControlPlane AddonProfiles cannot be completely removed",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AddonProfiles: []AddonProfile{
+						{
+							Name:    "first-addon-profile",
+							Enabled: true,
+						},
+						{
+							Name:    "second-addon-profile",
+							Enabled: true,
+						},
+					},
+					Version: "v1.18.0",
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AddonProfiles: []AddonProfile{
+						{
+							Name:    "first-addon-profile",
+							Enabled: true,
+						},
+					},
+					Version: "v1.18.0",
+				},
+			},
+			wantErr: true,
+		},
```

Comment:
Should we have a test spec for setting an AddonProfile to disabled (or is there already one)?

Reply:
There is one https://github.com/spectrocloud/cluster-api-provider-azure/blob/fix-addon-profile-upgdate/api/v1beta1/azuremanagedcontrolplane_webhook_test.go#L1060-#L1084